### DISCUSSION TITLE
Open File Picker in current directory of open buffer

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#11537] [Ideas] [answered]
I think this feature would be very useful if possible. In Telescope File Browser I can set a keymap to open the file picker in the path of the currently open buffer.

I do not see that option currently:
![image](https://github.com/user-attachments/assets/ca6ec81a-50ff-4801-88b6-91e4339700a7)

Here is the github for the Telescope File Picker
https://github.com/nvim-telescope/telescope-file-browser.nvim?tab=readme-ov-file#usage

I can set this keymap:
```-- open file_browser with the path of the current buffer
vim.keymap.set("n", "<space>fb", ":Telescope file_browser path=%:p:h select_buffer=true<CR>")
```

**Comment by gabydd** [comment]
there is a command for this it's ``file_picker_in_current_buffer_directory``

**Reply by ghost** [reply]
Here is my workflow:
Navigate to directory of code. (```~/code/projectA``` for example)
Open file with ```hx settings.py```
I make some edits.
Now I need to edit a file in another directory but I am not sure where it is, so I open my file picker.
Pressing ```f``` will open ```~/code/projectA``` since that is where the settings.py file is
Pressing ```F``` will open ```~/code/projectA``` since that is where I first opened helix

The file I am looking for is in another directory, so I find it and open it and edit it. (```~/code/projectB``` for example)
I then want to look for another file so I again need my file picker.
Pressing ```f``` will open ```~/code/projectA``` since that is where the first settings.py file is
Pressing ```F``` will open ```~/code/projectA``` since that is where I first opened helix

It should technically open ```~/code/projectB``` since that is where I currently am editing a file.

**Reply by ghost** [reply]
I think the assumption is that everyone will always be working in the same directory when I do not feel that is the case for most real world engineers (not just the youtubers working on a single project)

**Reply by kirawi** [reply]
Helix is intended to be opened per-project and LSPs were already designed with that in mind. The command suggested does what you're looking for and you need to bind it to a key sequence.

**Reply by ghost** [reply]
but what would be the key sequence to get the directory of the current buffer and open the file picker in that directory

**Reply by kirawi** [reply]
It is not bound by default.

**Reply by ghost** [reply]
Right so your telling me I would need to bind it but there is no documentation that I am aware of to tell me how I would to bind to some custom OS level command (like some bash script to find the current directory of the open buffer or something) I just see the ability to bind to premade functions like this https://docs.helix-editor.com/remapping.html
![image](https://github.com/user-attachments/assets/c87a285c-92ac-4c54-8d7c-e3c8fb5f69dc)


These all look like prebuilt helix internal commands

**Reply by kirawi** [reply, marked as answer]
Yes, `file_picker_in_current_buffer_directory` is a built-in command

**Reply by ghost** [reply]
oh I missed the 'buffer' part I thought you were refering me to the command in the docs I will try that thank you

**Reply by kirawi** [reply]
The colon is only if you are using typeable commands (e.g. :wq). It should work if you remove it
